### DISCUSSION TITLE
methods for improving mesh quality

### DISCUSSION THREAD

**Discussion by japresa** [idaholab/moose#19615] [Q&A General] [answered]
I'm currently creating a geometry for my code with trelis/cubit. However, I'm stuck and am unable to improve the mesh quality at the corners of my geometry. (Code won't run due to this)

 I've tried using the pillowing and general refinement methods at the geometry's side-surfaces & at the walls of the holes near the poor quality sections. However, any mesh-improvement methods I try seem to move the deformities elsewhere.
Any tips on how to improve mesh quality?

![image](https://user-images.githubusercontent.com/79496775/145756010-0407a692-8344-4fb2-a3a0-14e65b27e551.png)

![image](https://user-images.githubusercontent.com/79496775/145755056-b2760cb0-f2b2-4e38-b101-73efd0fa85fa.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello 

You can use MOOSE to smooth the meshes. 
Once your mesh is loaded by a FileMeshGenerator, then use a smooth mesh generator
https://mooseframework.inl.gov/source/meshgenerators/SmoothMeshGenerator.html

Guillaume

**Reply by japresa** [reply]
Going to try this. If this doesn't work, are there any other methods you know of?

**Reply by GiudGiud** [reply]
if very few nodes (like 10 or so) are problematic, you can move them one by one with https://mooseframework.inl.gov/source/meshgenerators/MoveNodeGenerator.html

For a mesh like this one (array of cylinders in a hex lattice), you could probably re-generate it using the moose mesh generators in the Reactor module
https://mooseframework.inl.gov/modules/reactor/index.html
